allennlp/modules/__init__.py

```diff
@@ -15,7 +15,10 @@
 from allennlp.modules.seq2seq_encoders import Seq2SeqEncoder
 from allennlp.modules.seq2vec_encoders import Seq2VecEncoder
 from allennlp.modules.similarity_functions import SimilarityFunction
-from allennlp.modules.span_pruner import SpanPruner
+from allennlp.modules.pruner import Pruner
+# TODO(mattg): This line is to keep backwards compatibility.  I'm not sure how to give a
+# deprecation warning on using this import, but we should remove this around version 0.8.
+from allennlp.modules.pruner import Pruner as SpanPruner  # pylint: disable=reimported
```

Comment:
This doesn't actually preserve backward compatibility, e.g what if someone had `from allennlp.modules.span_pruner import ....`.

Reply:
Maybe this isn't a huge deal though, up to you.

Reply:
Yeah, I'm not sure there's a way to fix that.  I'm hoping that most people will have imported from the base module, which is what we typically do.  @joelgrus, do you know of a better way we can handle backwards compatibility with renaming modules like this?

Reply:
I think you can just leave the `span_pruner.py` there and import the `Pruner` as `SpanPruner`?

Reply:
That would also be a natural place for the depreciation warning.

Reply:
Then I have to have docs for it, because of our documentation check...  Ugh.  Ok.

Reply:
just add it to the `docs to ignore` dictionary

Reply:
Thanks, I had forgotten about that.